src/test/regress/flaky_tests.md

```diff
@@ -0,0 +1,300 @@
++ 8970011 | reference_table                        |               |               | localhost |    57637
+ (13 rows)
+```
+
+**Example fix**:
+
+```diff
+-ORDER BY logicalrelid, shardminvalue::BIGINT;
++ORDER BY logicalrelid, shardminvalue::BIGINT, nodeport;
+```
+
+
+### Expected logs but different order
+
+
+**Issue**: The logs in the regress output are displayed in a different order
+than what the output file shows
+
+**Fix**: It's simple: don't log these things during the test. There are two common
+ways of achieving this:
+1. If you don't care about the logs for this query at all, then you can change
+   the log `VERBOSITY`
+2. If these are logs of uninteresting commands created by
+   `citus.log_remote_commands`. But you care about some of the other remote
+   commands being as expected. Then you can use `citus.grep_remote_commands` to
+   only display the commands that you care about.
+
+
+
+**Example of issue 1**:
+```diff
+select alter_table_set_access_method('ref','heap');
+ NOTICE:  creating a new table for alter_table_set_access_method.ref
+ NOTICE:  moving the data of alter_table_set_access_method.ref
+ NOTICE:  dropping the old alter_table_set_access_method.ref
+ NOTICE:  drop cascades to 2 other objects
+-DETAIL:  drop cascades to materialized view m_ref
+-drop cascades to view v_ref
++DETAIL:  drop cascades to view v_ref
++drop cascades to materialized view m_ref
+ CONTEXT:  SQL statement "DROP TABLE alter_table_set_access_method.ref CASCADE"
+ NOTICE:  renaming the new table to alter_table_set_access_method.ref
+  alter_table_set_access_method
+ -------------------------------
+
+ (1 row)
+```
+
+**Example fix of issue 1**
+```diff
++\set VERBOSITY terse
+```
+
+**Example of issue 2**
+```diff
+SET citus.log_remote_commands TO ON;
+ -- should propagate to all workers because no table is specified
+ ANALYZE;
+ NOTICE:  issuing BEGIN TRANSACTION ISOLATION LEVEL READ COMMITTED;SELECT assign_distributed_transaction_id(0, 3461, '2022-08-19 01:56:06.35816-07');
+ DETAIL:  on server postgres@localhost:57637 connectionId: 1
+ NOTICE:  issuing BEGIN TRANSACTION ISOLATION LEVEL READ COMMITTED;SELECT assign_distributed_transaction_id(0, 3461, '2022-08-19 01:56:06.35816-07');
+ DETAIL:  on server postgres@localhost:57638 connectionId: 2
+ NOTICE:  issuing SET citus.enable_ddl_propagation TO 'off'
+ DETAIL:  on server postgres@localhost:57637 connectionId: 1
+-NOTICE:  issuing SET citus.enable_ddl_propagation TO 'off'
+-DETAIL:  on server postgres@localhost:xxxxx connectionId: xxxxxxx
+ NOTICE:  issuing ANALYZE
+ DETAIL:  on server postgres@localhost:57637 connectionId: 1
++NOTICE:  issuing SET citus.enable_ddl_propagation TO 'off'
++DETAIL:  on server postgres@localhost:57638 connectionId: 2
+ NOTICE:  issuing ANALYZE
+ DETAIL:  on server postgres@localhost:57638 connectionId: 2
+```
+
+**Example fix of issue 2**
+```diff
+ SET citus.log_remote_commands TO ON;
++SET citus.grep_remote_commands = '%ANALYZE%';
```

Comment:
Cool, I didn't know we have this GUC.

Reply:
Nice to hear, people not knowing about some of these kind of tricks are exactly why I'm writing this document.